File: swimming-contest-software/Model/src/main/java/protocol/requests/FindAllSwimmersDetailsForRaceRequest.java
package protocol.requests;

import domain.dtos.RaceDetailsDTO;

public class FindAllSwimmersDetailsForRaceRequest implements Request {

    private final RaceDetailsDTO raceDetailsDTO;

    public FindAllSwimmersDetailsForRaceRequest(RaceDetailsDTO raceDetailsDTO) {
        this.raceDetailsDTO = raceDetailsDTO;
    }

    public RaceDetailsDTO getRaceDetailsDTO() {
        return raceDetailsDTO;
    }
}

File: Model/Protocol/Requests/FindAllSwimmersDetailsForRaceRequest.cs
using System;
using Model.Domain.DTOs;

namespace Model.Protocol.Requests;

[Serializable]
public class FindAllSwimmersDetailsForRaceRequest : IRequest
{
    public FindAllSwimmersDetailsForRaceRequest(RaceDetailsDTO raceDetailsDto)
    {
        RaceDetailsDTO = raceDetailsDto;
    }

    public RaceDetailsDTO RaceDetailsDTO { get; }
}